### PR TITLE
update Dockerfile.konflux

### PR DESCRIPTION
**Pull request by MohammadiIram**

### CODE REVIEW

**Comment by MohammadiIram** [comment]
Hello @atheo89 @jiridanek,

Could you please review the changes I have made? I’ve made some minor changes to the Dockerfile to align with the midstream Dockerfile. 

Thanks!

cc: @harshad16

File: components/notebook-controller/Dockerfile.konflux [hunk 1]
```diff
@@ -10,31 +10,25 @@ ARG TARGETARCH
...
-WORKDIR /opt/rhods/odh-notebook-controller
+# Update building workdir
+WORKDIR /opt/rhods/notebook-controller
 
 ## Build the odh-notebook-controller
 USER root
 
 # Build
-RUN CGO_ENABLED=1 GOOS=${TARGETOS:-linux} GOARCH=${TARGETARCH} GO111MODULE=on go build -tags strictfipsruntime -a -mod=mod -o manager main.go
+RUN  CGO_ENABLED=1 GOOS=${TARGETOS:-linux} GOARCH=${TARGETARCH} go build -tags strictfipsruntime -a -o ./bin/manager main.go
```
Comment (jiridanek) [inline components/notebook-controller/Dockerfile.konflux:28]:
```suggestion
RUN CGO_ENABLED=1 GOOS=${TARGETOS:-linux} GOARCH=${TARGETARCH} go build -tags strictfipsruntime -a -o ./bin/manager main.go
```

Isn't that extra space a typo? I wouldn't sync _that_ ;p

Comment (jiridanek) [inline components/notebook-controller/Dockerfile.konflux:28]: > `-mod=mod`

I'm still uncertain about that, but leaving it out is ok with me

* https://github.com/red-hat-data-services/kubeflow/pull/131/files#r1807802102

File: components/odh-notebook-controller/Dockerfile.konflux [hunk 2]
```diff
@@ -9,32 +9,21 @@ ARG TARGETARCH
...
-# components/common
-COPY ${SOURCE_CODE}/common ./common
-
-# Copy the Go Modules manifests
-COPY notebook-controller /workspace/notebook-controller
-COPY common /workspace/common
-
-# cache deps before building and copying source so that we don't need to re-download as much
-# and so that source changes don't invalidate our downloaded layer
-RUN cd /workspace/notebook-controller && go mod download
```
Comment (jiridanek) [inline components/odh-notebook-controller/Dockerfile.konflux:26]: this is the usual strategy about minimizing container rebuild time when go.mod does not change; I'm not sorry to see it go, but I would not mind keeping it

we can restore it in the future ourselves, at some point; for now I would just let konflux migration proceed

File: components/odh-notebook-controller/Dockerfile.konflux [hunk 3]
```diff
@@ -43,12 +32,16 @@ RUN microdnf install -y shadow-utils &&\
 ## Create a non-root user with UID 1001
 RUN useradd --uid 1001 --create-home --user-group --system rhods
 
+## Set workdir directory to user home
 WORKDIR /home/rhods
 
-COPY --from=builder /workspace/notebook-controller/manager .
-COPY --from=builder /workspace/notebook-controller/third_party/license.txt third_party/license.txt
```
Comment (jiridanek) [inline components/odh-notebook-controller/Dockerfile.konflux:49]: was never particularly clear on the importance of this license.txt, probably ok to leave it out

File: components/odh-notebook-controller/Dockerfile.konflux
Comment (jiridanek) [inline components/odh-notebook-controller/Dockerfile.konflux:1]: Why isn't there a Konflux build link down below in the pr checks tab in GitHub? I thought there is some integration setup, and that all PRs will be automatically built.

**Review by jiridanek — APPROVED** [review]
It does build and it starts the same way as always in `podman run`, so
/lgtm

To build, I did just

```
podman build -f notebook-controller/Dockerfile.konflux .
podman build -f odh-notebook-controller/Dockerfile.konflux .
```

so there are no konflux-only things in the Dockerfiles, which is also good

**Comment by jiridanek** [comment]
/approve cancel

forgot I should not click github approve button, because that has different meaning in prow than what we want

**Comment by jiridanek** [comment]
??? I did cancel the approve quickly enough, so what happened ???

**Comment by atheo89** [comment]
Hi @MohammadiIram, why were these changes applied directly to rhoai2.16 instead of going through master first? 
Also, could you include some reference to the related changes in the description next time? It’s difficult to switch context and make assumptions about why these changes were introduced.

Overall looks minor changes and enhancements on the Dockerfile /lgtm/
Please open a PR on the master to include this changes as well.

**Pull request merged**